### PR TITLE
Modernize the tiptap-editor

### PR DESCRIPTION
**Pull request by jeremygarey**
Changes:
- Upgrades Tiptap to v2.1.10 under the hood
- Converts to Vite for dev and building
- Updates the design of the editor
- Adds some vitest unit tests
- Adds actions for CI testing and npm publishing - triggered on the push of a new version tag
  - npm-publish requires a NPM_TOKEN secret is added
  - I'm not sure what permissions should look like for this repository. We obviously don't want the ability for anyone to trigger the publish action.
- Appropriate README updates

### CODE REVIEW

**Comment by EyePulp** [comment]
Whoa... 
image
This is a lot of substantial work @jeremygarey -- nice job.   Let's see if we can pull in @AlecTroemel to put eyes on it.

**Comment by jeremygarey** [comment]
@EyePulp - Perfect. Great to hear! And thank you for adding that token!

File: README.md [hunk 1]
```diff
@@ -1,4 +1,6 @@
-vue component that contains our custom setup for the tiptap editor. Able to highlihght phrases and give reasons/fixes to the warnings
+# tiptap-editor
+
+Vue component that contains our custom setup for the tiptap editor. Able to highlihght phrases and give reasons/fixes to the warnings
```
Comment (AlecTroemel) [inline README.md:3]: highlihght => highlight

**Comment by AlecTroemel** [comment]
@EyePulp @jeremygarey what are your thoughts on moving the character count to the top right of the toolbar? My reasoning is that there was a lot of wasted space on both the top and bottom "bars"

image

**Comment by EyePulp** [comment]
@alec - can you show where the count normally shows?

File: src/tiptap-editor.vue [hunk 2]
```diff
@@ -1,86 +1,99 @@
...
+                >
+                    <circle r="10" cx="10" cy="10" fill="#e9ecef" />
+                    <circle
+                        r="5"
+                        cx="10"
+                        cy="10"
+                        fill="transparent"
+                        stroke="currentColor"
+                        stroke-width="10"
+                        :stroke-dasharray="`calc(${characterCountPercentage} * 31.4 / 100) 31.4`"
```
Comment (AlecTroemel) [inline src/tiptap-editor.vue:87]: it looks like for this to work in Safari & Firefox you need to provide a unit within `calc`. The suggestion below does fix it for firefox on my machine
```suggestion
                        :stroke-dasharray="`calc(${characterCountPercentage}px * 31.4 / 100) 31.4`"
```

**Comment by AlecTroemel** [comment]
> @alec - can you show where the count normally shows?

there's a fancy screen in the readme on this branch! https://github.com/mirusresearch/tiptap-editor/blob/29-modernize-repo/img/example.png

**Comment by jeremygarey** [comment]
> @EyePulp @jeremygarey what are your thoughts on moving the character count to the top right of the toolbar? My reasoning is that there was a lot of wasted space on both the top and bottom "bars"
> 
> image

@AlecTroemel - It definitely looks nice at the top! And saves quite a bit of space. I'm all for it if we want to go that route.

File: src/tiptap-editor.vue (hunk 2 above)
Comment (jeremygarey) [reply src/tiptap-editor.vue:87]: Good catch. Thanks!

**Comment by EyePulp** [comment]
I'm good with the shift to the upper right for char counts.

File: src/tiptap-editor.vue (hunk 2 above)
Comment (AlecTroemel) [reply src/tiptap-editor.vue:87]: note that I definitely did not "just know" this :rofl:  I just forgot to link to the stack overflow question I found

https://stackoverflow.com/questions/65849051/calc-not-working-with-stroke-dashoffset-in-safari-and-firefox

**Comment by AlecTroemel** [comment]
@jeremygarey we should upgrade the version in `package.json` to `2.0.0`, but other then that this LGTM!

**Pull request merged**